### DISCUSSION TITLE
Table-Of-Contents With URL CMS Validator Failing With Zip File Download

### DISCUSSION THREAD

**Discussion by jkHCBB** [CMSgov/price-transparency-guide-validator#100] [Q&A] [answered]
Hey everyone. I'm having issues with the table-of-contents schema.
Below I have the code I'm running in Powershell:

`cms-mrf-validator validate 2023-09-07_Diabetes---Endocrinology-Center-of-Ohio--Inc-_index.json v1.0.0 -o table_of_contents_results.txt -t table-of-contents
info  Running validator container...
info  Table of contents refers to these files:
info  == In-Network Rates ==
info  * https://mrftest.healthcarebluebook.com/DIABETES&ENDOCRINOLOGYCENTEROFOHIO/272423
info  == Allowed Amounts ==
info  * https://mrftest.healthcarebluebook.com/DIABETES&ENDOCRINOLOGYCENTEROFOHIO/281134
Would you like to validate these files? [y/n]: y
info  File: https://mrftest.healthcarebluebook.com/DIABETES&ENDOCRINOLOGYCENTEROFOHIO/272423
info  Beginning download...
info  Downloaded 100% of file (4860 bytes)
info  Download complete.
info  Running validator container...
info  File: https://mrftest.healthcarebluebook.com/DIABETES&ENDOCRINOLOGYCENTEROFOHIO/281134
info  Beginning download...
info  Downloaded 100% of file (142843 bytes)
info  Download complete.
info  Running validator container...`

The Validator then returns the TXT file with the following:

Input JSON is valid.
https://mrftest.healthcarebluebook.com/DIABETES&ENDOCRINOLOGYCENTEROFOHIO/272423 - in-network
Input is not a valid JSON
Error(offset 0): Invalid value.
https://mrftest.healthcarebluebook.com/DIABETES&ENDOCRINOLOGYCENTEROFOHIO/281134 - allowed-amounts
Input is not a valid JSON
Error(offset 0): Invalid value.

I know that both of the Json files file downloaded from these urls with pass each respective schema when irun the Json file directly from my local machine, but they dont not pass when im running them as a url or in the TOC schema. The files are downloaded as a ZIP with only one Json file in them. 

This is the Json file from the TOC page that directs to Allowed Amounts and Inn Network. 

`{"reporting_entity_name":"Diabetes & Endocrinology Center of Ohio, Inc.","reporting_entity_type":"health plan","reporting_structure":[{"reporting_plans":[{"plan_name":"Trustmark HB","plan_id_type":"ein","plan_id":"351846036","plan_market_type":"group"}],"allowed_amount_file":{"location":"https://mrftest.healthcarebluebook.com/DIABETES&ENDOCRINOLOGYCENTEROFOHIO/281134","description":"Out of Network"}},{"reporting_plans":[{"plan_name":"202783588","plan_id_type":"ein","plan_id":"202783588","plan_market_type":"group"}],"in_network_files":[{"location":"https://mrftest.healthcarebluebook.com/DIABETES&ENDOCRINOLOGYCENTEROFOHIO/272423","description":"In Network"}]}],"FriendlyName":"Diabetes & Endocrinology Center of Ohio, Inc.  Table of Contents","BaseFileName":"Diabetes---Endocrinology-Center-of-Ohio--Inc-_index.zip","BaseFileNameShort":"Diabetes---Endocrinology-Center-of-Ohio--Inc-","plan_id":"_null_","plan_name":"_null_"}
`
Is anyone else having issues with the URL side of validator? Or am I just messing up something?

**Comment by mint-thompson** [comment, marked as answer]
Thank you for reporting this! This is a bug with how zip/gzip contents are detected when the initial request is redirected. I've opened #101 to address this bug.

**Reply by jkHCBB** [reply]
Thank you so much for the reply! I'll watch the 101 bug and continue after the update goes live.

**Reply by mint-thompson** [reply]
Version v1.5.2 has just been published to NPM. Please give it a try and let me know if everything proceeds as expected.

**Reply by jkHCBB** [reply]
Thank you again for the fast response time. Everything seems to be working as expected now. Kudos for the speedy turn around time. You all are awesome!
